### PR TITLE
PGP Encrypt using ResolveThenIdentify2

### PR DESCRIPTION
**Pull request by patrickxb**
@maxtaco 

also verified that this fixes https://keybase.atlassian.net/browse/CORE-2329 as well.

### CODE REVIEW

File: go/engine/pgp_encrypt.go [hunk 1]
```diff
@@ -184,56 +183,22 @@ func (e *PGPEncrypt) loadSelfKey() (*libkb.PGPKeyBundle, error) {
...
-		var err error
-		var username string
-		if loggedIn && !e.arg.SkipTrack {
-			username, err = e.trackUser(ctx, userAssert)
-		} else {
-			username, err = e.identifyUser(ctx, userAssert)
+		arg := keybase1.Identify2Arg{
+			UserAssertion: userAssert,
+			Reason: keybase1.IdentifyReason{
+				Type: keybase1.IdentifyReasonType_ENCRYPT,
```
Comment (maxtaco) [inline go/engine/pgp_encrypt.go]: Maybe set `AlwaysBlock : true` ?

Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: what does `AlwaysBlock` mean?  this is the same arg that DeviceKeyfinder uses (and thus saltpack encrypt).

Comment (maxtaco) [inline go/engine/pgp_encrypt.go]: To not short-circuit.  In this case, we'll get a weird U/I if we short-cirtcuit.

Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: So should it change for DeviceKeyfinder too?

Comment (maxtaco) [inline go/engine/pgp_encrypt.go]: I'd say yes.  There's a ticket for this, in which the U/I was weirdly interrupted before the end of the experience.

Comment (maxtaco) [inline go/engine/pgp_encrypt.go]: Let me know when you make this change, I'll take it for a spin.

Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: Changed (both places).

**Comment by maxtaco** [comment]
Took it for a test drive. LGTM!

**Pull request merged**